libs/common/files.cpp

```diff
@@ -14,32 +14,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-#include <dirent.h>
-#include "logger/logger.hpp"
+#include <boost/filesystem.hpp>
 
 #include "common/files.hpp"
+#include "logger/logger.hpp"
 
 void iroha::remove_all(const std::string &dump_dir) {
   auto log = logger::log("common::remove_all");
-  if (!dump_dir.empty()) {
-    // Directory iterator:
-    struct dirent **namelist;
-    auto status = scandir(dump_dir.c_str(), &namelist, nullptr, alphasort);
-    if (status < 0) {
-      log->error("Internal error on scanning folder {}", dump_dir);
+  try {
+    if (not boost::filesystem::exists(dump_dir)) {
```

Comment:
`boost::filesystem::exists` might throw an exception, try to make code exception-free.
You can use boost error code to replace exceptions
Check this out http://www.boost.org/doc/libs/1_53_0/libs/filesystem/doc/reference.html#exists